# Data Profiling Notebook

## Setup and Imports

In [5]:
# Automatically reload our modules if we change them in VSCode/PyCharm
%load_ext autoreload
%autoreload 2

# Add the project root to the path so we can import from src/
import sys
from pathlib import Path
sys.path.append(str(Path.cwd().parent.parent))

from src.data.loader import load_raw_data
from src.utils.profiler import generate_data_profile

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


## Load the Data

In [6]:
# Load dataset
df_titanic = load_raw_data("Housing.csv")

print(f"Dataset Shape: {df_titanic.shape}")
print(f"Total Duplicate Rows: {df_titanic.duplicated().sum()}")
df_titanic.head(3)

Dataset Shape: (545, 13)
Total Duplicate Rows: 0


,price,area,bedrooms,bathrooms,stories,mainroad,guestroom,basement,hotwaterheating,airconditioning,parking,prefarea,furnishingstatus
0,13300000,7420,4,2,3,yes,no,no,no,yes,2,yes,furnished
1,12250000,8960,4,4,4,yes,no,no,no,yes,3,no,furnished
2,12250000,9960,3,2,2,yes,no,yes,no,no,2,yes,semi-furnished


## Generate Data Profile

In [8]:
# Run our custom profiler
profile = generate_data_profile(df_titanic)
profile

,Data Type,Non-Null Count,Missing Values,Missing Percentage (%),Unique Values
price,int64,545,0,0.0,219
area,int64,545,0,0.0,284
bedrooms,int64,545,0,0.0,6
bathrooms,int64,545,0,0.0,4
stories,int64,545,0,0.0,4
mainroad,object,545,0,0.0,2
guestroom,object,545,0,0.0,2
basement,object,545,0,0.0,2
hotwaterheating,object,545,0,0.0,2
airconditioning,object,545,0,0.0,2


## Phase 2 Conclusion: Cleaning Plan

**Findings:**
- The dataset contains 545 rows and 13 columns.
- There are **0 missing values** and **0 duplicate rows**.
- Several features are categorical strings (e.g., `yes`/`no` or furnishing status).

**Action Plan for Phase 3 (Cleaning):**
- **No missing value treatment is required.**
- *Architectural Note:* While categorical variables (`mainroad`, `furnishingstatus`, etc.) need to be converted to numerical/binary formats for machine learning, this falls under **Feature Engineering**, not Data Cleaning. We will defer One-Hot Encoding and binary mapping to the ML preprocessing pipeline to maintain a strict separation of concerns. 
- The dataset will be passed as-is to the processed folder for EDA.